libs/sdk-ui-loaders/src/dashboard/dashboardLoader.tsx

```diff
@@ -0,0 +1,181 @@
+
+    private constructor(config: DashboardLoaderConfig) {
+        this.config = config;
+    }
+
+    public static dev(): DashboardLoader {
+        return new DashboardLoader({
+            engineLoader: staticDashboardEngineLoader,
+            pluginLoader: noopDashboardPluginLoader,
+        });
+    }
+
+    public static prod(): DashboardLoader {
+        // TODO: implement runtime-loaders & integrate them here
+        throw new Error("not implemented");
+    }
+
+    public onBackend = (backend: IAnalyticalBackend): IDashboardLoader => {
+        this.baseProps.backend = backend;
+
+        return this;
+    };
+
+    public fromClientWorkspace = (clientWorkspace: IClientWorkspaceIdentifiers): IDashboardLoader => {
+        this.clientWorkspace = clientWorkspace;
+        this.baseProps.workspace = undefined;
+
+        return this;
+    };
+
+    public fromWorkspace = (workspace: string): IDashboardLoader => {
+        this.baseProps.workspace = workspace;
+        this.clientWorkspace = undefined;
+
+        return this;
+    };
+
+    public forDashboard = (dashboardRef: ObjRef): IDashboardLoader => {
+        this.baseProps.dashboardRef = dashboardRef;
+        return this;
+    };
+
+    public withAdditionalPlugins = (...plugins: IDashboardPlugin[]): IDashboardLoader => {
+        this.additionalPlugins = plugins ?? [];
+
+        return this;
+    };
+
+    public withBaseProps = (props: IDashboardBaseProps): IDashboardLoader => {
+        this.baseProps = { ...props };
+
+        return this;
+    };
+
+    private resolveWorkspace = async (
+        backend: IAnalyticalBackend,
+    ): Promise<[string | undefined, IClientWorkspaceIdentifiers | undefined]> => {
+        if (!this.clientWorkspace) {
+            return [this.baseProps.workspace, undefined];
+        }
+
+        const resolvedClientWorkspace = await resolveLCMWorkspaceIdentifiers(backend, this.clientWorkspace);
+        invariant(
+            !isEmpty(resolvedClientWorkspace),
+            "DashboardLoader was not able to resolve LCM client workspace to actual workspace.",
+        );
+
+        return [resolvedClientWorkspace.workspace, resolvedClientWorkspace];
+    };
+
+    public load = async (): Promise<DashboardLoadResult> => {
+        const { backend, dashboardRef } = this.baseProps;
+
+        invariant(backend, "DashboardLoader is not configured with an instance of Analytical Backend.");
+        invariant(dashboardRef, "DashboardLoader is not configured with dashboard to load.");
+
+        const [workspace, clientWorkspace] = await this.resolveWorkspace(backend);
+        invariant(workspace, "DashboardLoader is not configured with workspace to use and loader.");
+
+        const dashboard = await backend.workspace(workspace).dashboards().getDashboard(dashboardRef);
+        const { engineLoader, pluginLoader } = this.config;
+
+        const engine = await engineLoader(dashboard);
```

Comment:
Detail: these two could be run in parallel.

Reply:
changed